boards/native/board_init.c

```diff
@@ -20,7 +20,57 @@
 
 #ifdef MODULE_MTD
 #include "mtd_native.h"
+
+mtd_native_dev_t mtd0_dev = {
+    .dev = {
+        .driver = &native_flash_driver,
+        .sector_count = MTD_SECTOR_NUM,
+        .pages_per_sector = MTD_SECTOR_SIZE / MTD_PAGE_SIZE,
+        .page_size = MTD_PAGE_SIZE,
+    },
+    .fname = MTD_NATIVE_FILENAME,
+};
+
+mtd_dev_t *mtd0 = (mtd_dev_t *)&mtd0_dev;
+#endif
+
+#ifdef MODULE_VFS
+#include "vfs.h"
+
+/*
+ * On `native` we define auto-mounts for every file system.
+ *
+ * A 'real' board would typically always use the same file system to avoid
+ * data loss when re-formatting, but since `native` is for testing only we
+ * provide all file system definitions here.
+ */
+
+/* littlefs support */
+#if defined(MODULE_LITTLEFS)
+
+#include "fs/littlefs_fs.h"
+VFS_AUTO_MOUNT(littlefs, &mtd0_dev, "/");
```

Comment:
Here and bellow, could `mtd0` be used instead?

Reply:
No 

```
In file included from /home/benpicco/dev/RIOT/boards/native/board_init.c:38:
/home/benpicco/dev/RIOT/sys/include/vfs.h:243:16: error: initializer element is not constant
  243 |         .dev = (mtd_dev_t *)(mtd),          \
      |                ^
/home/benpicco/dev/RIOT/boards/native/board_init.c:58:1: note: in expansion of macro ‘VFS_AUTO_MOUNT’
   58 | VFS_AUTO_MOUNT(littlefs2, mtd0, "/");
      | ^~~~~~~~~~~~~~
/home/benpicco/dev/RIOT/sys/include/vfs.h:243:16: note: (near initialization for ‘fs_desc.dev’)
  243 |         .dev = (mtd_dev_t *)(mtd),          \
      |                ^
/home/benpicco/dev/RIOT/boards/native/board_init.c:58:1: note: in expansion of macro ‘VFS_AUTO_MOUNT’
   58 | VFS_AUTO_MOUNT(littlefs2, mtd0, "/");
      | ^~~~~~~~~~~~~~
```